python/cudf/cudf/core/column/categorical.py

```diff
@@ -465,6 +465,18 @@ def reorder_categories(
         )
 
 
+def validate_categorical_children(children) -> None:
+    if not (
+        len(children) == 1
+        and isinstance(children[0], cudf.core.column.numerical.NumericalColumn)
+        and children[0].dtype.kind in "iu"
+    ):
+        # TODO: Enforce unsigned integer?
```

Comment:
Does it need to be? This would be removing the i in quotes above, I assume.

Reply:
I guess it doesn't strictly need to be since libcudf will probably do that casting. And yeah we would just remove the `i`